### DISCUSSION TITLE
Importer does not get "future" transactions from Nordigen

### DISCUSSION THREAD

**Discussion by Kr0kspluish** [firefly-iii/firefly-iii#8160] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I use a credit card, so the booking date of my card transactions are in the future (last day of next month). Usually, they show up normally, but this month, the transactions that should be booked for 2023-12-28 aren't imported. Actually, they seem to not even be returned by Nordigen. All transactions that are booked for 2023-11-30 are correctly imported.

### Debug information

Debug information generated at 2023-11-14 10:05:09 for Firefly III version **v6.0.30**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>v6.0.30 / v2.0.11 / 21 (exp. 21)</td></tr><tr><td>PHP version</td><td>8.2.12 / apache2handler / Linux</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>67108864 (64 MB)</td></tr><tr><td>Database drivers</td><td><strong>*mysql*</strong>, pgsql, sqlite, </td></tr><tr><td>Docker build</td><td><span>#</span>751, base <span>#</span>40</td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>Europe/Paris + Europe/Paris</td></tr><tr><td>App environment</td><td>local, debug: false</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>warning, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>fr_FR + equal</td></tr><tr><td>Trusted proxies</td><td>**</td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>2023-11-14 03:33:02 (6 hours ago)</td></tr><tr><td>Mailer</td><td>log</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td><span title="Has piggy banks">:pig:</span> <span title="Has rules">:wrench:</span> <span title="Has recurring transactions">:clock130:</span> <span title="Has object groups">:bookmark_tabs:</span> <span title="Has subscriptions">:email:</span></td></tr><tr><td>Session start</td><td>2023-12-01 09:36:38</td></tr><tr><td>Session end</td><td>2023-12-31 09:36:38</td></tr><tr><td>View range</td><td>1M</td></tr><tr><td>User language</td><td>fr_FR</td></tr><tr><td>User locale</td><td>fr_FR</td></tr><tr><td>Locale(s) supported</td><td>fr_FR.utf8: :white_check_mark:<br>fr_FR.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:109.0) Gecko/20100101 Firefox/119.0</td></tr></tbody></table>

### Expected behaviour

The recent credit card transaction should be imported with "bookingDate=2023-12-28"

### Additional info

It feels like this might be a Nordigen issue, but I'd like to make sure before reaching out to them directly.  

My auto-import config:
```JSON
{
    "add_import_tag": true,
    "roles": [],
    "do_mapping": [],
    "mapping": [],
    "duplicate_detection_method": "classic",
    "ignore_duplicate_lines": false,
    "unique_column_index": 0,
    "unique_column_type": "external-id",
    "flow": "nordigen",
    "content_type": "csv",
    "custom_tag": "",
    "identifier": "0",
    "connection": "0",
    "ignore_spectre_categories": false,
    "grouped_transaction_handling": "",
    "use_entire_opposing_address": false,
    "map_all_data": false,
    "accounts": {
        "XXX-XXXX-XXX": 1,
        "XXX-XXXX-XXX": 1
    },
    "date_range": "partial",
    "date_range_number": 7,
    "date_range_unit": "d",
    "date_not_before": "2023-10-01",
    "date_not_after": "",
    "nordigen_country": "FR",
    "nordigen_bank": "AGRICOLE_CENTRE_EST_AGRIFRPPXXX",
    "nordigen_requisitions": {
        "XXX-XXXX-XXX": "XXX-XXXX-XXX"
    },
    "nordigen_max_days": "30",
    "conversion": false,
    "ignore_duplicate_transactions": true
}
```

The beginning of the raw JSON string in the logs: 
```
[2023-11-14 09:45:10] production.DEBUG: JSON {"transactions":{"booked":[],"pending":[{"entryReference":"2331418349748","bookingDate":"2023-11-30","transactionAmount":{"amount":"-33.14","currency":"EUR"},"remittanceInformationUnstructuredArray":["CARTE","INTER GREECE 40 69 LYON 7E"]},
```

**Comment by JC5** [comment]
> Actually, they seem to not even be returned by Nordigen. 

Then there's not much I can do I'm afraid. I don't submit date limits or anything.

**Comment by Kr0kspluish** [comment]
That's what I feared. If you want updates, I can keep this issue to report Nordigen's answer when I contact them.

**Comment by JC5** [comment]
I'll convert it into a discussion, let me know when you hear anything!
